scripts/utils/consensus-celltype-plotting-functions.R

```diff
@@ -261,6 +261,127 @@ create_celltype_summary <- function(
 }
 
 
+#' Prep data frame to use for creating stacked bar plots showing immune cell types
+#'   with an emphasis on T and myeloid cell types 
+#'
+#' @param celltype_files List of files containing consensus cell type results
+#' @param all_immune_celltypes Vector of all consensus immune cell types possible
+#' @param tcell_celltypes Vector of T cell types to plot 
+#' @param myeloid_celltypes Vector of myeloid cell types to plot 
+#' @param frac_immune_threshold Threshold of fraction of immune cells in library required to include a library in the figure
+#'
+#' @returns Summarized data frame for input to plotting
+create_immune_celltype_summary <- function(
+    celltype_files,
+    all_immune_celltypes, 
+    tcell_celltypes, 
+    myeloid_celltypes, 
+    frac_immune_threshold
+){
+  
+  # read in consensus files and create data frame
+  consensus_df <- celltype_files |> 
+    purrr::map(readr::read_tsv) |> 
+    dplyr::bind_rows()
+  
+  # Determine libraries to remove as those with < frac_immune_threshold
+  #  fraction of immune cells out of total
+  remove_libraries <- consensus_df |>
+    # Keep only the immune cells and remove PDX
+    dplyr::filter(
+      sample_type == "patient tissue"
+    ) |>
+    dplyr::mutate(
+      category = ifelse(
+        consensus_annotation %in% c(tcell_celltypes, myeloid_celltypes), 
+        "immune", 
+        "other"
+      )
+    )  |>
+    dplyr::count(library_id, category) |>
+    tidyr::pivot_wider(names_from = category, values_from = n, values_fill = 0) |>
+    dplyr::rowwise() |>
+    dplyr::mutate(frac_immune = immune / (other + immune)) |>
+    dplyr::filter(frac_immune < frac_immune_threshold) |>
+    dplyr::pull(library_id)
+  
+  # subset to only immune celltypes, and add column for plotting
+  consensus_df <- consensus_df |>
+    # Keep only the immune cells and remove PDX
+    dplyr::filter(
+      consensus_annotation %in% all_immune_celltypes, 
+      sample_type == "patient tissue"
+    ) |>
+    # Create broad_celltype_group label with value "other" when cells are not T or myeloid types
+    dplyr::mutate(
+      broad_celltype_group = ifelse(
```

Comment:
Very minor, but I don't think we want to call this `broad_celltype_group` because it's different from the `broad_celltype_group` that's being used in 5B.